roles/openshift_storage_glusterfs/tasks/glusterfs_common.yml

```diff
@@ -85,6 +85,37 @@
 - include_tasks: glusterfs_deploy.yml
   when: glusterfs_is_native
 
+- name: Create heketi block secret
+  oc_secret:
+    namespace: "{{ glusterfs_namespace }}"
+    state: present
+    name: "heketi-{{ glusterfs_name }}-admin-secret-block"
+    type: "gluster.org/glusterblock"
+    force: True
+    contents:
+    - path: key
+      data: "{{ glusterfs_heketi_admin_key }}"
+  when:
+  - glusterfs_heketi_admin_key is defined
+  - glusterfs_block_storageclass
+
+- name: Generate Gluster Block StorageClass file
+  template:
+    src: "{{ openshift.common.examples_content_version }}/gluster-block-storageclass.yml.j2"
+    dest: "{{ mktemp.stdout }}/gluster-block-storageclass.yml"
+  when:
+  - glusterfs_block_storageclass
+
+- name: Create Gluster Block StorageClass
+  oc_obj:
+    state: present
+    kind: storageclass
+    name: "glusterfs-{{ glusterfs_name }}-block"
+    files:
+    - "{{ mktemp.stdout }}/gluster-block-storageclass.yml"
+  when:
+  - glusterfs_block_storageclass
+
```

Comment:
This whole chunk should be moved after the include of `glusterblock_deploy.yml`. No sense creating a StorageClass that can't provide storage yet. Also, wrap the tasks in a `block` segment and just apply the `glusterfs_block_storageclass` clause to the whole block. This will make it easier when we go to integrate Logging and Metrics deployments.

Reply:
Will do